File: Airsoft/src/app/models/products/guns/AllGunViewModel.ts
import { ImageViewModel } from "../../image/imageViewModel"

export interface AllGunViewModel {
    id: number,
    dealerName: string,
    name: string,
    manufacturer: string,
    weight: number,
    propulsion: string,
    power: number,
    color: string,
    price: number,
    image: ImageViewModel
}

File: serverAPI/AirsoftServer/ViewModels/Item/Guns/AllGunViewModel.cs
namespace ViewModels.Item.Guns
{
    using ViewModels.Images;

    public class AllGunViewModel
    {
        public int Id { get; init; }

        public string DealerName { get; init; }

        public string Name { get; init; }

        public string Manufacturer { get; init; }

        public double Weight { get; init; }

        public string Propulsion { get; init; }

        public double Power { get; init; }

        public string Color { get; init; }

        public decimal Price { get; init; }

        public ImageViewModel Image { get; init; }
    }
}
